pt/lessons/libraries/nimble-publisher.md

```diff
@@ -0,0 +1,347 @@
+
+```elixir
+defmodule NotFoundError, do: defexception [:message, plug_status: 404]
+
+def get_post_by_id!(id) do
+  Enum.find(all_posts(), &(&1.id == id)) ||
+    raise NotFoundError, "post with id=#{id} not found"
+end
+
+def get_posts_by_tag!(tag) do
+  case Enum.filter(all_posts(), &(tag in &1.tags)) do
+    [] -> raise NotFoundError, "posts with tag=#{tag} not found"
+    posts -> posts
+  end
+end
+```
+
+## Disponibilizando seu conteúdo
+
+Agora que já temos uma maneira de obter todas as nossas postagens e tags, as disponibilizar significa apenas conectar as rotas, controllers, views e templates da forma usual. Para esse exemplo iremos manter a simplicidade e apenas listar todas as postagens e obter uma postagem por seu id. É deixado a você como um exercício listar postagens por tags e estender o layout com as postagens recentes.
+
+### Rotas
+
+Defina as seguintes rotas em `lib/nimble_school_web/router.ex`:
+
+```elixir
+scope "/", NimbleSchoolWeb do
+  pipe_through :browser
+
+  get "/blog", BlogController, :index
+  get "/blog/:id", BlogController, :show
+end
+```
+
+### Controller
+
+Defina um controller para expor as postagens em `lib/nimble_school_web/controllers/blog_controller.ex`:
+
+```elixir
+defmodule NimbleSchoolWeb.BlogController do
+  use NimbleSchoolWeb, :controller
+
+  alias NimbleSchool.Blog
+
+  def index(conn, _params) do
+    render(conn, "index.html", posts: Blog.all_posts())
+  end
+
+  def show(conn, %{"id" => id}) do
+    render(conn, "show.html", post: Blog.get_post_by_id!(id))
+  end
+end
+```
+
+### View
+
+Crie o módulo view onde você pode colocar os auxiliares necessários para renderizar a view. Por agora é só:
+
+```elixir
+defmodule NimbleSchoolWeb.BlogView do
+  use NimbleSchoolWeb, :view
+end
+```
+
+### Template
+
+Finalmente, crie os arquivos HTML para rederizar o conteúdo. Dentro de `lib/nimble_school_web/templates/blog/index.html.eex` defina o seguinte para renderizar a lista de postagens:
+
+```html
+<h1>Listing all posts</h1>
+
+<%= for post <- @posts do %>
+<div id="<%= post.id %>" style="margin-bottom: 3rem;">
+  <h2><%= link post.title, to: Routes.blog_path(@conn, :show, post)%></h2>
+
+  <p><time><%= post.date %></time> by <%= post.author %></p>
+
+  <p>Tagged as <%= Enum.join(post.tags, ", ") %></p>
+
+  <%= raw post.description %>
+</div>
+<% end %>
+```
+
+E crie `lib/nimble_school_web/templates/blog/show.html.eex` para renderizar uma única postagem:
+
+```html
+<%= link "← All posts", to: Routes.blog_path(@conn, :index)%>
+
+<h1><%= @post.title %></h1>
+
+<p><time><%= @post.date %></time> by <%= @post.author %></p>
+
+<p>Tagged as <%= Enum.join(@post.tags, ", ") %></p>
+
+<%= raw @post.body %>
+```
+
+### Navegue por suas postagens!
+
+Tudo pronto para ir!
+
+Abra o servidor web com `iex -S mix phx.server` e visite [http://localhost:4000/blog](http://localhost:4000/blog) para conferir seu novo blog em ação!
```

Comment:
Sei que é a tradução da postagem em inglês mas fiquei me perguntando o porque ele iria iniciar o servidor com um IEx, só que logo depois ele pede para iniciar um sessão do IEx, o que esse ponto aqui acaba sendo mais factível se usar apenas a chamada para iniciar o server.

Reply:
@herminiotorres acho que vale levantar esse questionamento no texto original. Aqui seria interessante nos mantermos o mais fiéis possivel ao texto original. Oq acha?

Reply:
Eu entendi como preferência da pessoa que escreveu, daí escolhi por não alterar.

Reply:
Posso abrir no texto original, mas gostaria saber de vocês se faz sentido levar essa discussão a frente...
Pensando agora talvez seja muito :nail_care: nitpicking

Reply:
Por sinal, era sempre uma dúvida que eu sempre tive... se quem iria puxar as modificações sempre seriam as lições em inglês, ou alguma outra poderia soltar um novo release e a do inglês iria ficar desatualizado...

Reply:
@herminiotorres No geral toda modificação precisa entrar primeiro no inglês, e depois traduzida para cada idioma.

No [translation report](https://elixirschool.com/pt/report/) dá para ver o quão sincronizado com o inglês a tradução em português está.

Reply:
Legal saber :) 
Eu fiz o restante das alterações e vou resolver esse comentário aqui, tudo bem pessoas?